### ISSUE TITLE
Natural HP regeneration when sleeping seems too high?

### ISSUE THREAD

**Issue by Regularitee** [CleverRaven/Cataclysm-DDA#25547]
So, a human being can fully heal the most grievous HP losses in a single day of sleep, even without bionic, mutation, or medical assistance. In fact, medicine is largely useless for restoring HP in a timely fashion (except perhaps on characters that have reduced sleep traits, like Tireless).

Am I alone in thinking it cheapens the importance of avoiding injury? This is further exacerbated by the fact HP injuries do not significantly reduce player performance on their own, (but rather pain does), so it's entirely possible to work productively at crafting when on the verge of death.

I personally would really like to see the baseline healing factor reduced by a significant portion, so the worst HP losses take more than a day to fully recover from.

**Comment by sfsworms** [comment]
Agreed. Right now, the main use I have for disinfectant/bandage is late game when you can go on for days without sleep (thanks, coffee). Maybe some balance could be reached where sleeping while bandaged and disinfected would heal at roughly the current rate, while otherwise it might take a few days to shrug off a near-death injury. 

That'd still be very fast by real-world standard, but I 'm afraid that nerfing healing too much would make large part of the game into tediously waiting for the PC to heal.

**Comment by FulcrumA** [comment]
Fluff explanation was that it's one of the few subtle benefits of being infected by the blob. Kinda working as intended.

**Comment by Vasyan2006** [comment]
You can change default healing rate in `game_balance.json`. I prefer to play with reduced 10 times healing.

**Comment by latogato** [comment]
I'm sure soon or later the healing rate will be adjusted to the 91 days season. With 14 days long season + blob infection i think it is ok.

**Comment by sfsworms** [comment]
Healing rate was never set with regard to the 14 day season. The issue here is that with the shift to healing over time, there is not much benefits to bandages: if you get banged up, you won't heal from bandages before you go to bed and sleep will cure you anyway.

**Comment by puyebu** [comment]
I disagree with the huge reduction as I like the explanation that it is one of the blob infection benefits. Plus I don't really want tedium of healing for too long.
However, it would be not bad to have a system where more severe wounds take more time to heal, i.e. 10% -> 20% takes longer than 90% -> 100%. In my opinion it should make player care more about wounds but not make it too bothersome.

**Comment by latogato** [comment]
> Healing rate was never set with regard to the 14 day season.
Yeah, many of us know from the 91 days season long discussion.... Still fast  healing is acceptable with 14 days long season. If it would be realistic we would die or heal for seasons long.

> would be not bad to have a system where more severe wounds take more time to heal
I like it, the healing rate could be the same percentage as the current HP percentage to the max HP. So if the limb has 16/80 HP than it would be just 20% healing rate. It would give a good reason to use bandages.

**Comment by Regularitee** [comment]
The main issue is the wide range of possible max HP levels, and all the confounding variables. A mechanism must take into consideration possible combinations of healing accelerating or slowing effects: devices, mutations, bionics, medicines, and so on.

I think the best solution in the short term is a global drop in healing rate across the board. This would help prevent healing mutations, bionics, and medicine from being functionally useless (since normal humans already fully heal after sleep). 

In the longer term, the more realistic and detailed gradient healing that was suggested could be fine-tuned and developed before release.

**Comment by teapot156** [comment]
Even with the slowest healing rate it's still very fast

**Comment by nexusmrsep** [comment]
I also agree that it is too fast. It doesn't have to nerfed drastically but if you are on your last legs with red HP bars all over your body you shouldn't just heal overnight (even bandaged).

There must be incentive to use medical items, mutations etc. and sustained wound must not be trivial to heal to promote cautiousness instead of YOLO.

With my test character I exploded a grenade in hand while wearing RM13 armor so character survived, but was in severe pain and badly damaged all over. It had fast healer mutation from previous tests, and guess what? He healed faster then his natural pain reduction rate. That is not right.

**Comment by FulcrumA** [comment]
>but if you are on your last legs with red HP bars all over your body you shouldn't just heal overnight

Isn't it already the case? Though frankly, healing severe damage in two nights isn't that much better.

>There must be incentive to use medical items, mutations etc. and sustained wound must not be trivial to heal to promote cautiousness instead of YOLO.

I suspect that's where actual limb breaking comes in - if someone is not responsible with their injuries and let them get even worse, they will face consequences.

>It had fast healer mutation from previous tests, and guess what? He healed faster then his natural pain reduction rate. That is not right.

Here I agree. That does seem a bit iffy.

**Comment by aawilson** [comment]
All a drop in healing rate will do is make it more annoying to be injured (not difficult, just annoying). Retreat to your hidey hole and sleep and eat stored food for a week instead of a day. Without major changes to put more time pressure on not gathering resources, all you're doing is making the game more tedious.

**Comment by woutershep** [comment]
I've been playing a broken cyborg lately and am really struggling to stay alive thanks to that leaky bionic that wrecks my health and the damaging acid/scratching. Now that healing from items isnt instant anymore going to sleep heavily bandaged and disinfected is only enough to slow my inevetable death, broken cyborgs do not heal from a red bar to a full bar. Dropping the healing rate across the board sounds absolutely terrible for the viability of broken cyborgs.

**Comment by PenceGL** [comment]
I definitely agree that there needs to be a reason to use healing items, which is to me the main driving factor behind why reducing sleeping healing effectiveness makes a ton of sense.

As it is now, with the Repair Nanobots CBM combined with sleeping, I have no reason to use anything else. Even bandages aren't really needed any more. I think that the current systems in place are actually pretty good, they just need to be separated more. 

If we assign an effectiveness value to the current healing systems, they are all too close... let's use a scale of 1-10, with 1 being natural healing rate while awake and 10 being instant full healing. Everything we have right now feels pretty much centered around 5, with the exception of the RX12 Jet Injector which I think is still instant heal and heals a pretty good bit. It can have a 7.

Just off the top of my head, the goal separation would be something like

- Sleep with no support items = 1.5
- Sleep + bandage = 2
- Sleep + medical gauze (treated with antibiotics or something maybe?) = 3
- Repair nanobots = 6
- RX12  = 8

Of course this is subjective, it's just how it all feels to me. With separating the effectiveness more it opens up the possibility for different systems in between... now we could add something that is effectiveness value 4 or 5 that's slightly better than bandages, but still slightly more obtainable/accessible than the CBM. Or maybe it doesn't. Maybe this just makes it that much more rewarding/adds a large step in progression when you finally do jump up from medically treated bandages to nanobots.

**Comment by Regularitee** [comment]
I would advocate a **major** increase in energy usage for repair nanobots. Firstly, they are the ultimate end-game HP restore tool. At that point the player should have a large energy banks already (or if they don't, it will restrict the repair nanobots usage to prevent it from being used too often in early/mid game). And secondly, having a steep energy cost would still encourage players with it to still use medicine as much as possible, even while they have repair nanobots. But arguably this is getting off-topic, since this is an issue for baseline healing and not one specific CBM.

Back on topic, it seems the majority consensus is that unaugmented natural healing is too powerful. The question is, what should it be? Currently the game_balance.json indicates:

`    "type": "EXTERNAL_OPTION",`
`    "name": "PLAYER_HEALING_RATE",`
`    "info": "Set base player healing rate per turn. Default: 0.01",`
`    "stype": "float",`
`    "value": 0.01`

What should it be then, precisely? Please suggest specific values so we can submit a concrete PR change. After a new natural healing rate is decided upon, we can then re-visit the HP restoration of healing items (to provide at least some possibility of taking care of outlier cases like broken cyborgs ), since their ideal potency would depend on what the new healing rate is.

**Comment by aawilson** [comment]
> It isn't more tedious if it takes the same IRL time. The change is that injuries cost a week of food and food spoilage instead of a day's worth. Which puts more pressure on gathering resources and avoiding injury.

Why would it take the same IRL time? Because hitting the sleep key, then the eat/drink keys a few times, then the rest key over and over is instantaneous? So the player just gets the joy of wasting a bunch of time every time they bump into a hulk from slightly too close, or get bad luck fighting a shocker? Or do they get the joy of spending IRL hours repeatedly and tediously firing expensive ammunition at a wall until they finally can hit something from range?

**Comment by Vasyan2006** [comment]
>Now that healing from items isnt instant anymore going to sleep heavily bandaged and disinfected is only enough to slow my inevetable death, broken cyborgs do not heal from a red bar to a full bar.

Removing default health regeneration is a part of the brocken cyborg challenge.

>The question is, what should it be?

I preffer 0.001 for slow pace game. For default character healing will be 0.6 HP/hour or 4.8 HP after 8 hours of sleeping, instead of 48 HP before. Character will regenerate but not fast enought, and using medicine is required after every combat.

>we can then re-visit the HP restoration of healing items (to provide at least some possibility of taking care of outlier cases like broken cyborgs )

Healing from items is not related to this value. It depends on item power to apply healing effects and healing rate of these effects can be changed in effects.json. 
Worst health stat (cyborg or junkie) is reducing healing from items twice, so it is possible for them to get healing even if common regeneration is disabled.

It is planned to remove limb HP entirely and heal every wound separately:
https://github.com/CleverRaven/Cataclysm-DDA/projects
>A long-backburnered goal of mine is to replace HP with a distinct enumeration and interaction with wounds. This also includes overhauling "block", "dodge" and the effects of armor to tend to mitigate attacks rather than eliminate them.

**Comment by woutershep** [comment]
> Removing default health regeneration is a part of the brocken cyborg challenge.

Well duh. What im trying to say is that we should be carefull not to make the already hard cyborg challenge even harder or even outright impossible by nerfing the already low regenergation into the ground for low health characters. In other words, perhaps consider only changing the healing curve on the slightly-bad to good health range.

> Worst health stat (cyborg or junkie) is reducing healing from items twice, so it is possible for them to get healing even if common regeneration is disabled.

You seems to miss the fact that a cyborg continuesly takes damage. Currently this 'healing' at bad health levels is not enough to overcome the continual damage and leads to effectively not healing at all but instead just dying slower. Say hypotethically i can keep my cyborg alive for four days before it wastes away. If you reduce the healing by a factor 10 instead of having four days to cut that leaky bionic out of me i suddenly have 9.6 hours because i waste away 10 times as fast. Ofcourse super simplified and inacurate but i hope it gets the point across that im trying to make which you seem to miss.

So long story short i am looking forward to this change but to keep broken cyborg viability at the same level i can see three options, whichever is picked i dont care much but one has to do something to account for it if they want the difficulty to remain roughly the same:
1) Don't touch the healing curve for bad health characters too much.
2) Change healing items afterwards to compensate as per Regularitee's thought.
3) Change the bad bionics afterwards to make health/hp drain less bad to compensate.

**Comment by Vasyan2006** [comment]
@woutershep 
You want to balance HP regen based on one starting scenario.  I prefer the opposite, balance HP regen for different situations and change scenarios and bionics according to this. Well, actually I am not going to do it, because I have json settings for my characters and it is fine.
Brocken cyborg was heavily nerfed about two years ago when "health stat -> HP regen" relation was introduced. I suggest to open new issue about it or propose a solution as PR. But as I see characters with lowest health stat should be barely able to replenish HP.

**Comment by woutershep** [comment]
That's not what im saying at all and id prefer it if you didn't put words in my mouth like that. Everything i said here just condenses to 'Try to not leave a broken cyborg challenge in your wake after you fix this.'. Thats it. Nothing more. And i am gonna say nothing more on it now.

**Comment by teapot156** [comment]
> That's not what im saying at all and id prefer it if you didn't put words in my mouth like that. Everything i said here just condenses to 'Try to not leave a broken cyborg challenge in your wake after you fix this.'. Thats it. Nothing more. And i am gonna say nothing more on it now.

That's what you said. You backpedaled into this which does make sense:
"3. Change the bad bionics afterwards to make health/hp drain less bad to compensate."

One scenario needs balancing

**Comment by aawilson** [comment]
What problem does this actually solve? Is it some misguided worry about "realism" in a game with zombies and fusion guns and Morloss berries?

Consider where difficulty comes from in this game:

In the beginning, lots of stuff is deadly, and it is "easy" to overextend and be killed, either because you got surrounded by zombies, ran into a hulk with low stats and while already tired, or got unlucky on food spawns and starved to death. The only thing lower health changes is the last one, and here's what happens: You get to choose to lose quickly because you have to forage around while damaged, or you lose slowly by starving to death while you wait to heal.

After the beginning, food supplies quickly become a non-issue. Death usually comes from attempting "endgame" stuff too early, because there is always something to hunt and some way to cook it unless you really screwed up in a way that probably killed you earlier. A change to this, then, just means a slight miscalculation, rather than ending a trip early and postponing the rest by a day, postpones it by a week, a week of sitting around meticulously just doing the boring survival stuff you've already demonstrated you can do, just without any of the fun stuff to punctuate it. Nothing gets harder, because you'll still wait until healed to finish up whatever you were doing.

So, which of those sounds like it's more appealing, when a change like this is introduced?

**Comment by Night-Pryanik** [comment]
> "realism" in a game with zombies and fusion guns and Morloss berries

If you can't imagine realism **plus** zombies and fusion guns and Marloss berries (i.e. fantastic elements in otherwise realistic setting), that's entirely your problem. Introducing one fantastic element in the game doesn't means that we must automatically reject all other realistic elements.

**Comment by latogato** [comment]
> It is planned to remove limb HP entirely and heal every wound separately

I love it forward, it sound so unreal world like. I hate the limb HP system, we are like the Monty Python's black knight: "- oh, my left arm off after a short battle?  'Tis but a scratch.  I had worse. Come on you pansy! - proceed to swing at the zombie hulk with one arm."

**Comment by Endovior** [comment]
I can sympathize with this point of view; as is, HP damage is easily healed with nothing but bed rest... often in a single night! From a tedium-reduction point of view, that's still a desirable state for properly-equipped survivors, but I'd argue that it shouldn't be free.

What if we moved some of the current HP regeneration into medical items, such as bandages? That would mean that you can still expect to heal most wounds with a good night's sleep, but you'll need to actually spend some time and resources dealing with those wounds, instead of just ignoring them.

**Comment by Night-Pryanik** [comment]
> Calm your nuts, chico

I'm completely calm.

> this is an appropriate way to argue for it?

What inappropriate did you see in my post?

> Is it "realistic" that damage is treated like one big mass of abstract damage that all goes away entirely by itself? Is it "realistic" that none of those small cuts are subject to the normal infections that everyone in the world has to worry about even without zombies around? Is it "realistic" that there's no concept of internal hemorrhaging so hospital requirements can be conveniently handwaved?

You're making a big mistake here. We're slowly but steadily moving to the features you mentioned. These things weren't implemented yet because no one bothered about them, not because this is a specific design decision. Also features like that frequently encounter a big negative reaction from a lot of players, this is further delaying their implementation. Though I still hope sometime they will be in the game.

**Comment by aawilson** [comment]
> What inappropriate did you see in my post?

I quoted it. You're disregarding the contrary position as being a matter of not being able to imagine a "realistic" zombie survival game, calling it "my problem", which is hardly a counterargument.

>You're making a big mistake here. We're slowly but steadily moving to the features you mentioned.

Sure. And the problem raised in this thread makes no progress toward them. Again:

> why is this change the one that meaningfully moves the needle for that, and how does it make the game more fun for anyone but a small population of people in a game who could just choose a slow healer trait anyway and get some points for the trouble?

**Comment by Asmageddon** [comment]
I personally agree that it's overkill. I think that regenerating full health should take two nights worth of rest, rather than one.

For drawbacks... what about a "Soreness" debuff, with small penalties to speed, moderate to STR and AGI, and rare pain? Or perhaps -4 or so morale per stack?
